packages/api/authentication/handlers.go

```diff
@@ -0,0 +1,172 @@
+package authentication
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/gorilla/sessions"
+	"github.com/jiulongw/siwe-go"
+)
+
+var Store = sessions.NewCookieStore([]byte(os.Getenv("SESSION_KEY")))
+
+// seconds
+const MAX_NONCE_AGE = 30
+
+func IsAuthenticated(session *sessions.Session) bool {
+	return session.Values["siwe"] != nil
```

Comment:
```suggestion
	return session.Values["wallet"] != nil
```

Lets rename `siwe` on the cookie to `wallet` and serialize the db model (or perhaps just the id/wallet address). Additionally, we shouldn't access session/Store directly. we can encapsulate all that logic in the session middleware file (made those suggestions below too)